### ISSUE TITLE
Add config file 

### ISSUE THREAD

**Issue by stringhandler** [tari-project/sha-p2pool#16]
The config file should be implemented as a TOML file with the following settings:
1. Seed peers - an array of multiaddr for peers to be dialed. (also called bootstrap nodes)
2. Pool specific:
  1. Name of pool
  2. Number of shares
  3. Algorithm (sha3x for now)

**Comment by stringhandler** [comment]
Note that there is a standard way of doing config files in the Tari repo. This can help with using ENVs and command line option overriding, but other suggestions are also welcome

**Comment by therealdannzor** [comment]
I've been leaning towards building on existing config utilities, both for unity among projects but also usefulness of existing functionality, but there's an issue with versioning between the two.

`sha-p2pool` uses a newer version of Clap than `minotari_app_utilities`, and it could be argued to be a bit of regression for the project should we downgrade to the same version. Specifically, it concerns the struct [CommonCliArgs](https://github.com/tari-project/tari/blob/development/applications/minotari_app_utilities/src/common_cli_args.rs#L29) which is necessary since it already implements the trait [ConfigOverrideProvider](https://github.com/tari-project/tari/blob/development/common/src/configuration/mod.rs#L71).

I looked at bumping the Clap version on the `minotari_app_utilities` side to see if it was a quick fix in order for both of the projects to be on par. I noticed there were multiple new errors though, including annotated fields with attributes that I need look closer at exactly how they affect the rest of the module.

Suggest two ways of moving forward:
1. Schedule work to upgrade Clap in the `minotari_app_utilities` crate to unblock import in `sha-p2pool`
2. Implement a custom (likely simpler, without override) config system, specifically for `sha-p2pool`

Depends on the timeline we have on this feature, I'll put this on hold for now to focus on higher-priority protocol research.
